memote/suite/cli/reports.py

```diff
@@ -217,19 +226,29 @@ def diff(models, filename, pytest_args, exclusive, skip, solver,
         config.merge(ReportConfiguration.load(custom))
     # Build the diff report specific data structure
     diff_results = dict()
+    loaded_models = list()
     for model_path in models:
         try:
             model_filename = os.path.basename(model_path)
             diff_results.setdefault(model_filename, dict())
             model = callbacks._load_model(model_path)
             model.solver = solver
-            _, diff_results[model_filename] = api.test_model(
-                model, results=True, pytest_args=pytest_args,
-                skip=skip, exclusive=exclusive, experimental=experimental)
+            loaded_models.append(model)
         except Exception as e:
             LOGGER.warning(
                 "The following exception occurred while loading model {}: {}"
                 "".format(model_filename, e))
+    # Running pytest in individual processes to avoid interference
+    partial_test_diff = partial(_test_diff, pytest_args=pytest_args,
+                                skip=skip, exclusive=exclusive,
+                                experimental=experimental)
+    pool = Pool(2)
+    results = pool.map(partial_test_diff, loaded_models)
+
+    for model_path, result in zip(models, results):
```

Comment:
Can you really rely on the order of `models` and `results` being the same? Somehow I doubt that.